libs/api/metadata-converter/src/lib/dcat-ap/read-parts.ts

```diff
@@ -1,48 +1,78 @@
 import { NamedNode, Statement, Store } from 'rdflib'
 import { DCAT, DCTERMS, FOAF, LOCN, RDF, SKOS, VCARD } from './namespaces'
-import { findNodeLocalized } from './utils/graph-utils'
+import { readLocalizedValue } from './utils/graph-utils'
 import {
   DatasetDownloadDistribution,
   DatasetServiceDistribution,
   DatasetSpatialExtent,
   Individual,
   Keyword,
+  LanguageCode,
   OnlineLinkResource,
   OnlineResource,
+  OnlineResourceTranslations,
   Organization,
+  RecordTranslations,
 } from '@geonetwork-ui/common/domain/model/record'
 import { getAsValidUrl } from '../common/url'
 import { fullNameToParts } from '../iso19139/utils/individual-name'
 import { readLicenseFromString } from '../common/license'
 import { matchProtocol } from '../common/distribution.mapper'
+import { LANG_3_TO_2_MAPPER } from '@geonetwork-ui/util/i18n/language-codes'
 
-function getDataset(dataStore: Store, recordNode: NamedNode): NamedNode {
+function getDatasetNode(dataStore: Store, recordNode: NamedNode): NamedNode {
   return (dataStore.the(recordNode, FOAF('primaryTopic'), null) ||
     dataStore.the(null, RDF('type'), DCAT('Dataset'))) as NamedNode
 }
 
+function getCatalogNode(dataStore: Store, recordNode: NamedNode): NamedNode {
+  return (dataStore.the(null, DCAT('record'), recordNode) ||
+    dataStore.the(null, RDF('type'), DCAT('Catalog'))) as NamedNode
+}
+
 export function readUniqueIdentifier(
   dataStore: Store,
   recordNode: NamedNode
 ): string {
   return dataStore.the(recordNode, DCTERMS('identifier'), null)?.value
 }
 
-export function readTitle(dataStore: Store, recordNode: NamedNode): string {
-  const dataset = getDataset(dataStore, recordNode)
-  return findNodeLocalized(dataStore, dataset, DCTERMS('title'), null, 'en')
-    ?.value
+export function readTitle(
+  dataStore: Store,
+  recordNode: NamedNode,
+  translations: RecordTranslations,
+  defaultLanguage: LanguageCode
+): string {
+  const dataset = getDatasetNode(dataStore, recordNode)
+  const [title] = readLocalizedValue<RecordTranslations>(
```

Comment:
`readLocalizedValue` is introduced in this PR and does two things:
* add any found translations to the `translations` object given to it
* return both the default value and the translations array as a tuple: `[value, translations]`